smpclient/__init__.py

```diff
@@ -500,18 +506,18 @@ def _maximize_file_upload_packet(self, request: FileUpload, data: bytes) -> File
             len=request.len,
         )
 
-    async def _initialize(self) -> None:
+    async def _initialize(self, timeout_s: float | None = None) -> None:
         """Gather initialization information from the SMP server."""
+        timeout_s = timeout_s if timeout_s is not None else self._timeout
 
         try:
-            async with timeout(2):
-                mcumgr_parameters = await self.request(MCUMgrParametersRead())
-                if success(mcumgr_parameters):
-                    logger.debug(f"MCUMgr parameters: {mcumgr_parameters}")
-                    self._transport.initialize(mcumgr_parameters.buf_size)
-                elif error(mcumgr_parameters):
-                    logger.warning(f"Error reading MCUMgr parameters: {mcumgr_parameters}")
-                else:
-                    assert_never(mcumgr_parameters)
-        except asyncio.TimeoutError:
+            mcumgr_parameters = await self.request(MCUMgrParametersRead(), timeout_s=timeout_s)
+            if success(mcumgr_parameters):
+                logger.debug(f"MCUMgr parameters: {mcumgr_parameters}")
+                self._transport.initialize(mcumgr_parameters.buf_size)
+            elif error(mcumgr_parameters):
+                logger.warning(f"Error reading MCUMgr parameters: {mcumgr_parameters}")
+            else:
+                assert_never(mcumgr_parameters)
+        except TimeoutError:
```

Comment:
This bit is perhaps worth a change in the future. Technically, if this times out, it may indicate a situation in which the transport, but not the server, is available. Potential opportunity for fail fast, but also opportunity for confusing error messages on initialization due to a server that's slow to respond.